runtime/reference/ts_config_migration.md

```diff
@@ -0,0 +1,501 @@
+---
+title: "Configuring TypeScript"
+oldUrl:
+  - /runtime/manual/advanced/typescript/faqs/
+  - /runtime/manual/advanced/typescript/migration/
+  - /runtime/manual/advanced/typescript/configuration/
+  - /runtime/manual/advanced/typescript/types/
+  - /runtime/manual/typescript/types/
+  - /runtime/manual/advanced/faqs/
+---
+
+Deno’s flexibility shines in its equal treatment of TypeScript and JavaScript.
+Whether you’re transitioning from JavaScript to TypeScript or vice versa, Deno
+has features to ease the journey.
+
+## Type Checking JavaScript
+
+You may wish to make your JavaScript more type-sound without adding type
+annotations everywhere. Deno supports using the TypeScript type checker to type
+check JavaScript. You can mark individual files by adding the check JavaScript
+pragma to the file:
+
+```js
+// @ts-check
+```
+
+This will cause the type checker to infer type information about the JavaScript
+code and raise any issues as diagnostic issues.
+
+These can be turned on for all JavaScript files in a program by providing a
+configuration file with the check JS option enabled:
+
+```json
+{
+  "compilerOptions": {
+    "checkJs": true
+  }
+}
+```
+
+And then setting the `--config` option on the command line.
+
+## Using JSDoc in JavaScript
+
+When type-checking JavaScript or importing JavaScript into TypeScript, JSDoc
+annotations can provide additional type information beyond what can just be
+inferred from the code itself. Deno supports this seamlessly if you annotate
+your code inline with the supported
+[TypeScript JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html).
+
+For example to set the type of an array use the following JSDoc comment:
+
+```js
+/** @type {string[]} */
+const a = [];
+```
+
+## Skipping type checking
+
+You might have TypeScript code that you are experimenting with, where the syntax
+is valid but not fully type safe. You can bypass type checking for a whole
+program by passing the `--no-check` flag.
+
+You can also skip whole files being type checked, including JavaScript if you
+have check JS enabled, by using the `nocheck` pragma:
+
+```js
+// @ts-nocheck
+```
+
+## Renaming JS files to TS files
+
+While this might work in some cases, it has some severe limits in Deno. This is
+because Deno, by default, runs type checking in what is called _strict mode_.
+This means a lot of unclear or ambiguous situations where are not caught in
+non-strict mode will result in diagnostics being generated, and JavaScript is
+nothing but unclear and ambiguous when it comes to types.
+
+## Configuring TypeScript in Deno
+
+While TypeScript offers many configuration options, which can be daunting if
+you're just starting out with TS. Deno aims to simplify using TypeScript,
+instead of drowning you in countless settings, Deno configures TypeScript to
+**just work** out of the box. No extra configuration headaches required!
```

Comment:
The first sentence reads a bit odd:

> While TypeScript offers many configuration options, which can be daunting if
you're just starting out with TS.

Maybe we can start with the good parts:

```suggestion
TypeScript is configured out of the box as you'd expect with Deno. For most projects you'll never have to touch any TypeScript related configuration options. If you do run into the need to change some of the TypeScript's compiler options, you can do so via `deno.json`.
```

Reply:
Have updated